src/runtime_src/core/tools/xbmgmt2/ReportPlatform.cpp

```diff
@@ -190,11 +190,17 @@ ReportPlatform::getPropertyTree20202( const xrt_core::device * device,
   f.getBoardInfo(info);
   //create information tree for a device
   pt_platform.put("bdf", xrt_core::query::pcie_bdf::to_string(xrt_core::device_query<xrt_core::query::pcie_bdf>(device)));
-  pt_platform.put("flash_type", xrt_core::device_query<xrt_core::query::flash_type>(device)); 
-  pt_platform.put("hardware.serial_num", info.mSerialNum.empty() ? "N/A" : info.mSerialNum);
+  pt_platform.put("flash_type", xrt_core::device_query<xrt_core::query::flash_type>(device));
+  std::string sn = "N/A";
+  if(info.mSerialNum.empty())

```

Comment:
Can you please help me understand why `info.mSerialNum` doesn't work for devices with sdm?

Reply:
board_info is filled up using XMC driver's sysfs nodes. And, it doesn't work on versal devices.
So, need to add this case where serial_num is empty or "N/A" then look for this data in hwmon_sdm driver. This driver is added in place of XMC driver on versal device. All sensor information is parsed from this driver only but not from XMC driver.
I hope it clarifies why there is a separate query_requests for hwmon_sdm instead of using xmc ones.